torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -347,6 +354,198 @@ at::Tensor tensor_from_cuda_array_interface(PyObject* obj) {
       at::device(kCUDA).dtype(dtype)
   );
 }
+
+
+template <typename T>
+std::string get_vector_str(std::vector<T> & vec, int64_t start, int64_t end=-1){
+  std::string output = "(";
+  if (end < 0) {
+    end = vec.size();
+  }
+  for (int i=start; i<end; ++i) {
+    output += (std::to_string(vec[i]) + ", ");
+  }
+  return output + ")";
+}  
+
+template <typename T>
+std::string get_buffer_str(void* buffer, int64_t start, int64_t end){
+  T* array = static_cast<T*>(buffer); 
+  std::string output = "(";
+  for (int i=start; i<end; ++i) {
+    output += (std::to_string(array[i]) + ", ");
+  }
+  return output + ")";
+}  
+
+
+bool _extract_ndarrays(PyObject* obj, std::vector<int64_t> & shape, size_t (&numels)[], size_t shape_idx, std::vector<PyArrayObject*> & array_ptr_storage) {
+  if (PyArray_Check(obj)) {
+    if (numels[shape_idx] != PyArray_SIZE((PyArrayObject*) obj)) {
```

Comment:
Add a comment explaining what's happening in this block. The error logic really dominates.

Is it sufficient to just check the number of elements? Isn't the shape of the array relevant?

Reply:
I actually gave a bit of a leeway. As long as the number of elements are the same as the expected number of elements the results will be correct since the raw memory copy routines sees a numpy array and its flattened version in the same way.